src/native/eventpipe/ep-session.c

```diff
@@ -202,7 +203,7 @@ ep_session_alloc (
 	instance->session_start_time = ep_system_timestamp_get ();
 	instance->session_start_timestamp = ep_perf_timestamp_get ();
 	instance->paused = false;
-	instance->enable_stackwalk = ep_rt_config_value_get_enable_stackwalk ();
+	instance->enable_stackwalk = stacks_requested && ep_rt_config_value_get_enable_stackwalk ();
```

Comment:
@noahfalk, @davmason, based on the implementation of the global env variable in https://github.com/dotnet/runtime/pull/82470 and my comment on that PR, https://github.com/dotnet/runtime/pull/82470#issuecomment-1443085890, I believe it should be possible to turn off stack traces global using the new env variable introduced in https://github.com/dotnet/runtime/pull/82470, but then being able to re-enable it per session. Agree?

Reply:
Usually I feel like env vars are designed to trump other sources of configuration. So if someone told me there was an IPC command to configure per-session values and also an env var EventPipeEnableStackwalk=0, I would guess that the env var takes precedence whenever it is present. Certainly its possible not to implement it that way, but if session configuration takes precedence over the env var users might find the behavior surprising. 

My hope is that we could leave it as env-var takes precedence but very few users would ever need to use the env-var. Ideally scenarios where users wanted the env-var will change to be configured in the tool (we should add a switch to dotnet-trace) and then users configure their tools rather than configuring the env-var.

The scenario my suggestion doesn't handle would be a user that runs two or more tools against the same app, one of those tools supports the new IPC command and does need stacks and a 2nd tool does not use the new IPC command and the user doesn't want that one to have stacks. My guess is that this scenario will be rare enough that we are OK ignoring it. Does that seem reasonable or you are thinking this will be more common than I anticipate?

Reply:
OK, then I believe we probably need to adjust the naming of the parameter to make the intent more clearer. If the env variable should enable/disable stack walking for all sessions and not just being a default value, then the session parameter can only do one thing, disable stack walking (since stack walks are enabled by default) and if EventPipeEnableStackwalk=0 has been set, disabling stack walking in a session would just ends up being a noop, but the client will still get expected behavior, no stack walks on events emitted into that session. If we use current name suggestion for the parameter, request_stackwalk (true/false) and the runtime has been setup with EventPipeEnableStackwalk=0 you will get less intuitive behavior's where you request stack walks but will still end up with a session without stack walks due to env variable configuration, not visible for the client requesting the trace.

Reply:
In the past we've treated environment variables as niche things and haven't tried too hard to make them user friendly, I am ok with leaving it as is or with making the changes you describe @lateralusX.

Maybe this is a sign that I've been working in the runtime too long, but having sessions be able to request stacks and then the environment variable overriding it doesn't seem that bad. But, like I said I am good with either approach

Reply:
> Maybe this is a sign that I've been working in the runtime too long, but having sessions be able to request stacks and then the environment variable overriding it doesn't seem that bad

I feel similarly to David. We have many environment variables that override things and the naming of the settings elsewhere generally ignores the presence of the env vars. For example DOTNET_gcServer forces GC server mode, DOTNET_JITMinOpts disables all JIT optimizations, or DOTNET_TieredCompilation overrides whether tiered compilation is enabled.